Fix linux build

nevyn opened the pull request:
Some of the code was using Apple specifics. Removing them makes this library work fine on Linux.

nevyn committed:
Import Foundation for replaceOccurrences

This method is in Foundation, not in stdlib, so needs to be explicitly imported to be able to build on Linux.

nevyn committed:
On Linux, don't use Apple-specific intrinsics in Angle

they're apple specific extensions. Use glibc instead.

eugenebokhan commented:
@nevyn Thank you for the PR. Could you please make sure the CI checks pass, maybe add linux check alongside the Mac one?

nevyn committed:
Avoid importing Glibc unless needed

nevyn commented:
@eugenebokhan The code builds and tests successfully locally on my Mac. The build errors look like CI issue, perhaps bumping the Swift version will help?

There is no `#if os(linux)` in Swift, but perhaps it's the Glibc import that messes things up. I've conditionalized it, but I can't trigger CI from here.

nevyn committed:
Depend on kvSIMD on Linux

I was able to build this before because Swift 6.1 would just pick up kvSIMD from higher up in the dependency chain, which was wrong. In Swift 6.2, we need to explicitly depend on kvSIMD to have a chance to build.
